dixit/src/test/java/tw/wally/dixit/model/RoundTest.java

```diff
@@ -0,0 +1,251 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+import static java.util.List.of;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static tw.wally.dixit.model.Game.NUMBER_OF_PLAYER_HAND_CARDS;
+import static tw.wally.dixit.model.GameTest.DEFAULT_CARD_SIZE;
+import static tw.wally.dixit.utils.StreamUtils.generate;
+import static tw.wally.dixit.utils.StreamUtils.skip;
+
+public class RoundTest extends AbstractDixitTest {
+
+    private static final int NUMBER_OF_GUESSERS = 3;
+    private Round currentRound;
+    private Player storyteller;
+    private List<Player> guessers;
+    private Map<Player, Card> cardOfPlayers;
+
+    @BeforeEach
+    public void beforeTest() {
+        int numberOfPlayers = NUMBER_OF_GUESSERS + 1;
+        var players = generatePlayers(numberOfPlayers);
+        this.storyteller = players.get(0);
+        this.guessers = skip(players, 1);
+        this.currentRound = new Round(storyteller, guessers);
+        dealAllPlayersCards(players);
+        assertEquals(RoundState.STORY_TELLING, currentRound.getState());
+        this.cardOfPlayers = new HashMap<>(numberOfPlayers);
+    }
+
+    private void dealAllPlayersCards(Collection<Player> players) {
+        var deck = new LinkedList<>(generateCards(DEFAULT_CARD_SIZE));
+        players.forEach(player -> player.addHandCards(generate(NUMBER_OF_PLAYER_HAND_CARDS, index -> deck.pollLast())));
+    }
+
+    @Test
+    public void WhenStorytellerTellStory_ThenRoundStateShouldBeCardPlaying() {
+        tellStory();
+
+        assertEquals(RoundState.CARD_PLAYING, currentRound.getState());
+    }
+
+    @Test
+    public void WhenGuesserTellStory_ThenShouldFail() {
+        var guesser = guessers.get(0);
+
+        assertThrows(IllegalArgumentException.class, () -> tellStory(guesser));
+    }
+
+    @Test
+    public void GivenStoryTold_WhenStorytellerTellStoryAgain_ThenShouldFail() {
+        tellStory();
+
+        assertThrows(IllegalArgumentException.class, this::tellStory);
+    }
+
+    @Test
+    public void GivenStoryTold_WhenAllGuessersPlayCard_ThenRoundStateShouldBePlayerGuessing() {
+        tellStory();
+
+        guessers.forEach(this::playCard);
+
+        assertEquals(RoundState.PLAYER_GUESSING, currentRound.getState());
+    }
+
+    @Test
+    public void GivenStoryToldAndGuesser1PlayedCard_WhenGuesserAPlayCardAgain_ThenShouldFail() {
+        tellStory();
+        var Guesser1 = guessers.get(0);
+        playCard(Guesser1);
+
+        assertThrows(IllegalArgumentException.class, () -> playCard(Guesser1));
+    }
+
+    @Test
+    public void GivenStoryToldAndAllGuessersPlayedCard_WhenFirstGuesserPlayCardAgain_ThenShouldFail() {
+        givenStoryToldAndAllGuessersPlayedCard();
+
+        var firstGuesser = guessers.get(0);
+        assertThrows(IllegalArgumentException.class, () -> playCard(firstGuesser));
+    }
```

Comment:
In the last test case, you use `guesser1` instead of `firstGuesser`. Remember the ubiquitous language :D